compiler/rustc_resolve/src/lib.rs

```diff
@@ -3458,8 +3465,8 @@ fn module_to_string(module: Module<'_>) -> Option<String> {
     Some(names_to_string(&names))
 }
 
-#[derive(Copy, Clone, PartialEq, Debug)]
-enum CrateLint {
+#[derive(Copy, Clone, Debug)]
+enum RecordUsed {
```

Comment:
Maybe this enum should have some different name.
It should be short because it's used often.

There are two path resolution modes:
- Speculative mode - we only return the result, but do not report errors or lints, and do not do recovery
- Non-speculative mode - we report errors, do recovery when possible, also report lints, and mark path targets as used (the `record_used` naming comes from this). That's why need some extra data like a node id for linting or a span in this mode.

Maybe call it `enum Finalize`?
It's two letters shorter, not as specific as the old "record used" naming, and we use "finalize" elsewhere in resolve for similar functionality, like `fn finalize_(imports,macro_resolutions)`.

Reply:
Maybe something like
```rust
enum Mode {
  Speculative,
  Finalize
}
```
?

`enum Finalize` looks OK to me though.

Reply:
I've renamed everything to `Finalize`/`finalize`, but kept it a single flat enum (with `Finalize::No` variant), otherwise it becomes too annoying to both construct and unwrap it.